src/TypeSpec.Extension/Emitter.Csharp/src/lib/clientModelBuilder.ts

```diff
@@ -221,6 +223,30 @@ export function createModelForService(
     setUsage(usages, modelMap);
     setUsage(usages, enumMap);
 
+    // propagate the isConfident from the types to the operations
+    for (const client of clients) {
+        for (const op of client.Operations) {
+            let isOperationConfident = true;
+            for (const parameter of op.Parameters) {
+                // skip the special parameters
+                if (
+                    parameter.IsApiVersion ||
+                    parameter.IsEndpoint ||
+                    parameter.IsContentType
+                ) {
+                    parameter.Type.IsConfident = true; // we never recognize these as low confident level
```

Comment:
Why are we setting values in the propagation process?

Reply:
because they are constructed elsewhere separately

Reply:
This is not propagation process, per your suggestion, now propagation and calculation happen together when we constructing them.

These parameters are constructed elsewhere, and this is a kind of workaround because we still recognize the "union" of literal values as union types therefore sometimes, the content type will become "IsConfident: false"
This is a easy workaround of that issue - but in the meantime, let me see if I could resolve it from its root cause.

Reply:
This is resolved.